modules/virt-discovering-vm-internal-fqdn.adoc

```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+// * virt/vm_networking/virt-accessing-vm-internal-fqdn.adoc              
+
+:_mod-docs-content-type: PROCEDURE                                   
+[id="virt-discovering-vm-internal-fqdn_{context}"]
+= Connecting to a virtual machine by using its internal FQDN
+
+To connect to a virtual machine (VM) from within the cluster by using its internal FQDN, you must first map the VM to a headless service. Set the `spec.hostname` and `spec.subdomain` parameters in the VM configuration file. If a headless service exists with a name that matches the subdomain, a unique DNS A record is created for that VM in the form of `<vm.spec.hostname>.<vm.spec.subdomain>.<vm.metadata.namespace>.svc.cluster.local`.
+
+.Procedure
+
+. Edit the `VirtualMachine` manifest to add the service selector label and subdomain:
++
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm-fedora
+spec:
+  template:
+    metadata:
+      labels:
+        expose: me # <1>
+    spec:
+      hostname: "myvm" # <2>
+      subdomain: "mysubdomain" # <3>
+# ...
+----
+<1> The `expose:me` label must match the `spec.selector` attribute of the `Service` manifest that you previously created.
+<2> If this attribute is not specified, the resulting DNS A record takes the form of `<vm.metadata.name>.<vm.spec.subdomain>.<vm.metadata.namespace>.svc.cluster.local`.
+<3> The `spec.subdomain` attribute must match the `metadata.name` value of the `Service` object.
+
+. Save and exit the editor.
+
+. Restart the VM to apply the changes.
+
+. Run the following command to connect to the VM console:
++
+[source,terminal]
+----
+$ virtctl console vm-fedora
+----
+
+. Use the ping utility to find the DNS entry for the VM:
++
+[source,terminal]
+----
+$ ping myvm.mysubdomain
+----
++
+.Example output
+[source,terminal]
+----
+PING myvmi.mysubdomain.default.svc.cluster.local (10.244.0.57) 56(84) bytes of data.
+64 bytes from myvm.mysubdomain.default.svc.cluster.local (10.244.0.57): icmp_seq=1 ttl=64 time=0.029 ms
+----
++
+In the above example, the DNS entry for `myvm.mysubdomain.default.svc.cluster.local` points to `10.244.0.57`, which is the IP address of the VM.
```

Comment:
This will not work with VMs (although with Pods it would, so perhaps this is a bug.)

```suggestion
. Use the requested FQDN to connect to the VM:
+
[source,terminal]
----
$ ping myvm.mysubdomain.<namespace>.svc.cluster.local
----
+
.Example output
[source,terminal]
----
PING myvmi.mysubdomain.default.svc.cluster.local (10.244.0.57) 56(84) bytes of data.
64 bytes from myvm.mysubdomain.default.svc.cluster.local (10.244.0.57): icmp_seq=1 ttl=64 time=0.029 ms
----
+
In the above example, the DNS entry for `myvm.mysubdomain.default.svc.cluster.local` points to `10.244.0.57`, which is the cluster IP address currently assigned to the VM.
```

Reply:
I have updated the PR. @yossisegev Let me know if you are able to validate the procedure.